src/operations/multireweight.lisp

```diff
@@ -146,14 +164,29 @@ Then, we reach the \"critical segment\", where it becomes impossible to rewind p
           (loop :for reply :in replies :unless (null reply)
                 :do (assert (not (minusp (message-pong-weight reply))))
                     (setf internal-pong
-                          (unify-pongs internal-pong reply))))))))
+                          (unify-pongs internal-pong reply)))
+          #+i(format t "(~a) [~a] {~a}: internal-pong: ~a~%" (now) supervisor roots internal-pong)
+          (with-slots (target-root) internal-pong
```

Comment:
i think this continuation-packing is a little different from how we do it with other operators. should we put all these commands in up top + just set an aborting flag by this point?

Reply:
Yeah multireweight is a tricky one because it needs to do collect the hold cluster and do another scan before entering its critical section. Even before this change it differed from the rest of the operations. This is because the arguments to some of its commands (specifically `hold-cluster` and `targets`) are not known at supervisor spawn time, which is unlike every other operation.